frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.component.tsx

```diff
@@ -42,102 +47,135 @@ const emptyOption = prepareSingleContainerData({
 	role: '',
 	isFavourite: false,
 });
-const filter = createFilterOptions<IContainer>();
+const filter = createFilterOptions<IContainer>({ trim: true });
 
 export const UploadListItemDestination = ({
-	errorMessage,
+	control,
+	errors,
 	disabled = false,
 	className,
 	onChange,
 	defaultValue,
+	containersNamesInUse,
+	setContainersNamesInUse,
 	...props
 }: IUploadListItemDestination): JSX.Element => {
 	const [value, setValue] = useState<IContainer>({ ...emptyOption, name: defaultValue });
 	const [disableClearable, setDisableClearable] = useState(!value.name);
 	const containers = ContainersHooksSelectors.selectContainers();
-	const processingContainers = containers
-		.filter((container) => !canUploadToBackend(container.status));
+	const processingContainersNames = containers
+		.filter((container) => !canUploadToBackend(container.status))
+		.map(({ name }) => name);
 	const [newOrExisting, setNewOrExisting] = useState(() => {
 		if (value.name) {
 			return containers.find((c) => c.name === value.name) ? 'existing' : 'new';
 		}
-		return 'unset';
+		return '';
 	});
-
-	const [containersInUse, setContainersInUse] = useState(processingContainers);
 	const { getValues } = useFormContext();
 	const forceUpdate = useCallback(() => {
-		const containerIdsInModal = getValues().uploads.map((upload) => upload.containerId).filter(Boolean);
-		if (containerIdsInModal) {
-			const containersInModal = containerIdsInModal.map((idInUse) => containers
-				.find((cont) => cont._id === idInUse));
-			setContainersInUse([...processingContainers, ...containersInModal]);
+		const containerNamesInModal = getValues().uploads.map((upload) => upload.containerName.trim()).filter(Boolean);
+		if (containerNamesInModal) {
+			setContainersNamesInUse([...processingContainersNames, ...containerNamesInModal]);
 		}
 	}, []);
+	const federationsNames = FederationsHooksSelectors.selectFederations().map(({ name }) => name);
+
+	const errorMessage = errors.containerName?.message;
+
+	const onAutocompleteChange = (_, newValue: IContainer) => {
+		setValue(newValue || emptyOption);
+		onChange(newValue || emptyOption);
+		if (!newValue) {
+			setNewOrExisting('');
+			forceUpdate();
+		} else {
+			setNewOrExisting(!newValue._id ? 'new' : 'existing');
+		}
+		setDisableClearable(!newValue);
+	};
+
+	const onAutocompleteInputChange = (_, name: string) => {
+		if (name === value.name) onChange(value);
+	};
+
+	const getFilterOptions = (options: IContainer[], params) => {
+		let filtered: IContainer[] = filter(options, params);
+		const { inputValue } = params;
+
+		const isExisting = options.some((option: IContainer) => inputValue === option.name);
```

Comment:
I think inputValue needs to be trimmed otherwise it doesn't trigger isExisting, so typing a preexisting container with spaces on the end visually looks like you're adding a new container